query/stdlib/influxdata/influxdb/from.go

```diff
@@ -3,23 +3,576 @@ package influxdb
 import (
 	"fmt"
 
+	"github.com/influxdata/flux"
+	"github.com/influxdata/flux/ast"
 	"github.com/influxdata/flux/execute"
 	"github.com/influxdata/flux/plan"
+	"github.com/influxdata/flux/semantic"
 	"github.com/influxdata/flux/stdlib/influxdata/influxdb"
+	"github.com/influxdata/flux/stdlib/universe"
+	"github.com/influxdata/flux/values"
 	platform "github.com/influxdata/influxdb"
 	"github.com/influxdata/influxdb/query"
 	"github.com/pkg/errors"
 )
 
+const FromKind = "influxDBFrom"
+
+type FromOpSpec struct {
+	Bucket   string `json:"bucket,omitempty"`
+	BucketID string `json:"bucketID,omitempty"`
+}
+
 func init() {
-	execute.RegisterSource(influxdb.FromKind, createFromSource)
+	fromSignature := semantic.FunctionPolySignature{
+		Parameters: map[string]semantic.PolyType{
+			"bucket":   semantic.String,
+			"bucketID": semantic.String,
+		},
+		Required: nil,
+		Return:   flux.TableObjectType,
+	}
+
+	flux.ReplacePackageValue("influxdata/influxdb", influxdb.FromKind, flux.FunctionValue(influxdb.FromKind, createFromOpSpec, fromSignature))
```

Comment:
So for this replacement, you are passing the `influxdb.FromKind`, which is `"from"` and not `"influxDBFrom"`... should it be the latter?

Reply:
That is required to replace the actual call of `from`, so that it is overridden by InfluxDB's from. Otherwise, a caller should write:

```
import "influxdb"

influxdb.influxDBFrom(...) |> ...
``` 
What I think is incorrect is here:

```
flux.FunctionValue(influxdb.FromKind, createFromOpSpec, fromSignature)
```

I think it should be:

```
flux.FunctionValue(FromKind, createFromOpSpec, fromSignature)
```

I tried both ways and tests passes, maybe I am missing something.

Reply:
Would be good to get to the bottom of that.  I would like to understand it better too.

Reply:
Yeah, got to the bottom with @jlapacik and it turns out that the kind passed to `FunctionValue` is redundant and not used. The interpreter needs only to know that a certain id in a package is mapped to some function value that invokes a precise `createOpSpec`, no more. However, I'll change to 

```
flux.FunctionValue(FromKind, createFromOpSpec, fromSignature)
```

For consistency with other implementations.